sessionctx/variable/sysvar.go

```diff
@@ -1298,11 +1353,17 @@ var defaultSysVars = []*SysVar{
 	{Scope: ScopeGlobal | ScopeSession, Name: TiDBSlowLogMasking, Value: BoolToOnOff(DefTiDBRedactLog), Type: TypeBool, SetSession: func(s *SessionVars, val string) error {
 		// TiDBSlowLogMasking is deprecated and a alias of TiDBRedactLog.
 		return s.SetSystemVar(TiDBRedactLog, val)
+	}, SetGlobal: func(s *SessionVars, val string) error {
+		// TiDBSlowLogMasking is deprecated and a alias of TiDBRedactLog.
+		return s.GlobalVarsAccessor.SetGlobalSysVarOnly(TiDBRedactLog, val)
 	}},
 	{Scope: ScopeGlobal | ScopeSession, Name: TiDBRedactLog, Value: BoolToOnOff(DefTiDBRedactLog), Type: TypeBool, SetSession: func(s *SessionVars, val string) error {
 		s.EnableRedactLog = TiDBOptOn(val)
 		errors.RedactLogEnabled.Store(s.EnableRedactLog)
+		s.systems[TiDBSlowLogMasking] = val
```

Comment:
I've changed this to use `Aliases`. I don't think it's strictly necessary, but since this is a refactor I'd like to change that decision to a later PR that is clear in intentions.